docs/reference/intro.asciidoc

```diff
@@ -155,81 +184,146 @@ You can use a combination of dynamic and explicit mapping on the same index.
-similarity searches, and prefix searches, and get autocomplete suggestions.
+* <<full-text-queries,*Full-text search*>>. Search text that has been analyzed and indexed to support full-text search based on relevance.
+* <<keyword,*Keyword search*>>. Search for exact matches using `keyword` fields.
+* <<semantic-search-semantic-text,*Semantic search*>>. Search `semantic_text` fields using dense or sparse vector search on embeddings generated in your {es} cluster.
+* <<knn-search,*Vector search*>>. Search for similar dense vectors using the kNN algorithm for embeddings generated outside of {es}.
+* <<geo-queries,*Geospatial search*>>. Search for locations and calculate spatial relationships using geospatial queries.
 
-Have geospatial or other numerical data that you want to search? {es} indexes
-non-textual data in optimized data structures that support
-high-performance geo and numerical queries.
+Learn about the full range of queries supported by the <<query-dsl,Query DSL>>.
 
-You can access all of these search capabilities using {es}'s
-comprehensive JSON-style query language (<<query-dsl, Query DSL>>). You can also
-construct <<sql-overview, SQL-style queries>> to search and aggregate data
-natively inside {es}, and JDBC and ODBC drivers enable a broad range of
-third-party applications to interact with {es} via SQL.
+You can also filter data using the Query DSL.
+Filters enable you to include or exclude documents by retrieving documents that match specific field-level criteria.
+A query that uses the `filter` parameter indicates <<filter-context,filter context>>.
+
+{esql} also has powerful filtering capabilities.
 
 [discrete]
 [[analyze-data]]
 ==== Analyzing your data
 
-{es} aggregations enable you to build complex summaries of your data and gain
-insight into key metrics, patterns, and trends. Instead of just finding the
-proverbial “needle in a haystack”, aggregations enable you to answer questions
-like:
+{es} enables a host of use cases based on aggregations and analytics.
+These include real-world applications like:
 
-* How many needles are in the haystack?
-* What is the average length of the needles?
-* What is the median length of the needles, broken down by manufacturer?
-* How many needles were added to the haystack in each of the last six months?
+* *Log analysis*: Monitoring system performance and detecting anomalies in IT infrastructure.
+* *E-commerce analytics*: Analyzing sales trends, customer behavior, and product performance.
+* *Search analytics*: Tracking user search patterns and improving relevance of search results.
+* *Business intelligence*: Generating real-time dashboards and reports for data-driven decision making.
+* *Content recommendation*: Analyzing user interactions to provide personalized content suggestions.
 
-You can also use aggregations to answer more subtle questions, such as:
+[discrete]
+[[analyze-data-query-dsl]]
+===== Query DSL
```

Comment:
this heading comes out of nowhere. need to connect query languages with "analyzing your data" - maybe in headings

overall, I think you should make the relationships between the QLs and these sections more consistent.

Reply:
> overall, I think you should make the relationships between the QLs and these sections more consistent.

💯 

maybe focus on Query DSL and ES|QL, outlining search, filtering, and analysis capabilities for each

and then appendix with table that gives honorable mention to others